src/libawkward/array/ListOffsetArray.cpp

```diff
@@ -432,49 +432,66 @@ namespace awkward {
+      if (mindepth == depth  &&  maxdepth == depth) {
+        if (depth - 1 + axis < 0) {
+          throw std::invalid_argument(std::string("ListOffsetArrayOf<T> cannot be counted in axis ") + std::to_string(axis) + std::string(" because its depth is ") + std::to_string(depth));
+        }
+        return count(depth - 1 + axis);
+      }
+      else {
+        return content_.get()->count(axis);
+      }
     }
-    IndexOf<T> starts = make_starts(offsets_);
-    IndexOf<T> stops = make_stops(offsets_);
-    int64_t lenstarts = starts.length();
-    IndexOf<T> tocount(starts.length());
-    struct Error err = util::awkward_listarray_count(
-      tocount.ptr().get(),
-      starts.ptr().get(),
-      stops.ptr().get(),
-      lenstarts,
-      starts.offset(),
-      stops.offset());
-    util::handle_error(err, classname(), identities_.get());
-    std::vector<ssize_t> shape({ (ssize_t)lenstarts });
-    std::vector<ssize_t> strides({ (ssize_t)sizeof(T) });
-    std::string format;
+    else if (axis == 0) {
+      IndexOf<T> starts = make_starts(offsets_);
+      IndexOf<T> stops = make_stops(offsets_);
+      int64_t lenstarts = starts.length();
+      IndexOf<T> tocount(starts.length());
+      struct Error err = util::awkward_listarray_count(
+        tocount.ptr().get(),
+        starts.ptr().get(),
+        stops.ptr().get(),
+        lenstarts,
+        starts.offset(),
+        stops.offset());
+      util::handle_error(err, classname(), identities_.get());
+      std::vector<ssize_t> shape({ (ssize_t)lenstarts });
+      std::vector<ssize_t> strides({ (ssize_t)sizeof(T) });
+      std::string format;
 #ifdef _MSC_VER
-    if (std::is_same<T, int32_t>::value) {
-      format = "l";
-    }
-    else if (std::is_same<T, uint32_t>::value) {
-      format = "L";
-    }
-    else if (std::is_same<T, int64_t>::value) {
-      format = "q";
-    }
+      if (std::is_same<T, int32_t>::value) {
+        format = "l";
+      }
+      else if (std::is_same<T, uint32_t>::value) {
+        format = "L";
+      }
+      else if (std::is_same<T, int64_t>::value) {
+        format = "q";
+      }
 #else
-    if (std::is_same<T, int32_t>::value) {
-      format = "i";
-    }
-    else if (std::is_same<T, uint32_t>::value) {
-      format = "I";
-    }
-    else if (std::is_same<T, int64_t>::value) {
-      format = "l";
-    }
+      if (std::is_same<T, int32_t>::value) {
+        format = "i";
+      }
+      else if (std::is_same<T, uint32_t>::value) {
+        format = "I";
+      }
+      else if (std::is_same<T, int64_t>::value) {
+        format = "l";
+      }
 #endif
+      else {
+        throw std::runtime_error("unrecognized ListArray specialization");
+      }
+      return std::make_shared<NumpyArray>(Identities::none(), util::Parameters(), tocount.ptr(), shape, strides, 0, sizeof(T), format);
+    }
     else {
-      throw std::runtime_error("unrecognized ListArray specialization");
+      return content_.get()->count(axis);
```

Comment:
Whatever you do on new line 503 above, you need to do the _same thing_ here. I think the correct behavior is

```c++
return std::make_shared<ListOffsetArray>(Identities::none(), util::Parameters(), offsets_, content_.get()->count(axis - 1);
```

and similarly for ListArray. I _think_.

Reply:
done